apps/web/src/app/(private)/evaluations/[evaluationUuid]/editor/_components/EvaluationEditor/Editor/index.tsx

```diff
@@ -0,0 +1,108 @@
+'use client'
+
+import { Suspense, useCallback, useEffect, useState } from 'react'
+
+import { ConversationMetadata, readMetadata } from '@latitude-data/compiler'
+import { Evaluation, PARAMETERS_FROM_LOG } from '@latitude-data/core/browser'
+import {
+  AppLocalStorage,
+  DocumentTextEditor,
+  DocumentTextEditorFallback,
+  DropdownMenu,
+  useLocalStorage,
+} from '@latitude-data/web-ui'
+import useEvaluations from '$/stores/evaluationsStore'
+import { useDebouncedCallback } from 'use-debounce'
+
+import { Header } from './Header'
+
+export default function EvaluationEditor({
+  evaluation,
+}: {
+  evaluation: Evaluation
+}) {
+  const { updateEvaluation } = useEvaluations()
+  const [value, setValue] = useState(evaluation.prompt)
+  const [isSaved, setIsSaved] = useState(true)
+  const [metadata, setMetadata] = useState<ConversationMetadata>()
+
+  const { value: showLineNumbers, setValue: setShowLineNumbers } =
+    useLocalStorage({
+      key: AppLocalStorage.editorLineNumbers,
+      defaultValue: true,
+    })
+  const { value: wrapText, setValue: setWrapText } = useLocalStorage({
+    key: AppLocalStorage.editorWrapText,
+    defaultValue: true,
+  })
+  const { value: showMinimap, setValue: setShowMinimap } = useLocalStorage({
+    key: AppLocalStorage.editorMinimap,
+    defaultValue: false,
+  })
+
+  const debouncedSave = useDebouncedCallback(
+    (val: string) => {
+      updateEvaluation({
+        evaluationUuid: evaluation.uuid,
+        prompt: val,
+      })
+      setIsSaved(true)
+    },
+    500,
+    { trailing: true },
+  )
+
+  const onChange = useCallback((value: string) => {
+    setIsSaved(false)
+    setValue(value)
+    debouncedSave(value)
+  }, [])
+
+  useEffect(() => {
+    readMetadata({
+      prompt: value,
+      withParameters: Object.keys(PARAMETERS_FROM_LOG),
+    }).then(setMetadata)
+  }, [value])
```

Comment:
Currently, the Evaluation Editor is using `PARAMETERS_FROM_LOG` directly instead of the premade `readMetadataFromEvaluation` function. This is because this must be run directly from the browser, instead of being a service.

The compiling should be done in the server, though, as it will require running the AI.